MapboxNavigation/RouteVoiceController.swift

```diff
@@ -159,14 +161,24 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
     }
     
     func duckAudio() throws {
-        let categoryOptions: AVAudioSessionCategoryOptions = [.duckOthers, .interruptSpokenAudioAndMixWithOthers]
-        try AVAudioSession.sharedInstance().setMode(AVAudioSessionModeSpokenAudio)
-        try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: categoryOptions)
+        if #available(iOS 12.0, *) {
+            let categoryMode = CarPlayManager.isConnected ? AVAudioSessionModeVoicePrompt : AVAudioSessionModeSpokenAudio
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: categoryMode, options: categoryOptions)
+        } else if #available(iOS 10.0, *) {
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, mode: AVAudioSessionModeSpokenAudio, options: categoryOptions)
+        } else {
+            try AVAudioSession.sharedInstance().setMode(AVAudioSessionModeSpokenAudio)
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: categoryOptions)
+        }
         try AVAudioSession.sharedInstance().setActive(true)
     }
     
     func mixAudio() throws {
-        try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryAmbient)
+        if #available(iOS 12.0, *), CarPlayManager.isConnected {
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback, with: categoryOptions)
```

Comment:
Both `duckAudio()` and `mixAudio()` specify `AVAudioSessionCategoryOptionInterruptSpokenAudioAndMixWithOthers` and `AVAudioSessionCategoryOptionDuckOthers`. However, the names of these methods suggest that maybe one should only duck and the other should only mix. Would that be correct?

Reply:
again, is this really specific to CarPlay? We want to avoid creating a coupling here if at all possible.

Reply:
That's correct.